src/components/dependencies/Component.js

```diff
@@ -247,6 +282,19 @@ const Component = React.memo(({
       },
     }}
   >
+    {
+      (
+        mountRescoring
+        && (artefactNodes.length > 0)
+        && isParentComponent // only allow rescoring linking for root component
```

Comment:
nit: If it's the parent component, `artefactNodes` should be empty anyways so you could omit the check here.